### PR TITLE
[MSA] Redirect to home page for enabling 2FA

### PR DESCRIPTION
**Pull request by shishirx34**
Addressing a minor fix for redirecting to home page(for redirected pages after authentication) for encouraging 2FA.

### CODE REVIEW

File: src/NuGetGallery/Filters/UiAuthorizeAttribute.cs [hunk 1]
```diff
@@ -19,9 +19,12 @@ public UIAuthorizeAttribute(bool allowDiscontinuedLogins = false)
...
         {
-            // If the user has a discontinued login claim, redirect them to the homepage
+            // If the user has a discontinued login claim or should enable 2FA, redirect them to the homepage
             var identity = filterContext.HttpContext.User.Identity as ClaimsIdentity;
-            if (!AllowDiscontinuedLogins && ClaimsExtensions.HasDiscontinuedLoginClaims(identity))
+            var askUserToEnable2FA = filterContext.Controller?.TempData?.ContainsKey(GalleryConstants.AskUserToEnable2FA);
+
+            if ((!AllowDiscontinuedLogins && ClaimsExtensions.HasDiscontinuedLoginClaims(identity))
+                || (askUserToEnable2FA.HasValue && askUserToEnable2FA.Value))
             {
```
Comment (zhhyu) [inline src/NuGetGallery/Filters/UiAuthorizeAttribute.cs]: Do we have the test coverage for this change?

File: src/NuGetGallery/Filters/UiAuthorizeAttribute.cs [hunk 2]
```diff
@@ -19,9 +19,12 @@ public UIAuthorizeAttribute(bool allowDiscontinuedLogins = false)
 
         public override void OnAuthorization(AuthorizationContext filterContext)
         {
```
Comment (zhhyu) [inline src/NuGetGallery/Filters/UiAuthorizeAttribute.cs]: Null check for filterContext?

Comment (shishirx34) [reply src/NuGetGallery/Filters/UiAuthorizeAttribute.cs]: The attribute is only added to the routes that have the authorization context. This has been in play for over a year, the null check isn't necessary.

File: src/NuGetGallery/Filters/UiAuthorizeAttribute.cs (hunk 1 above)
Comment (shishirx34) [reply src/NuGetGallery/Filters/UiAuthorizeAttribute.cs]: added

**Pull request merged**